libqtile/scripts/launch.py

```diff
@@ -0,0 +1,59 @@
+# Copyright (c) 2025, elParaguayo
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the \"Software\"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED \"AS IS\", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""
+Command-line wrapper to launch process from within qtile
+"""
+
+import argparse
+import os
+import select
+
+
+def launch(opts) -> None:
+    fd = opts.fd
+
+    if fd > 0:
+        # Check if there's anything to read on the file descriptor
+        # but don't block
```

Comment:
Why not block? Isn't the point of this to block?

Reply:
Bad wording. It's a timeout so it blocks for a second. 

I included a timeout so that, if for some reason the message doesn't get sent, we still launch the required process.

Reply:
But if the message doesn't get sent, isn't that a bug? So with the timeout, we'll just be masking the bug, IIUC.

Reply:
You could look at it like that!

My thinking was that spawning the process should happen, regardless of whether we're create the rule or not. If we just try to read the socket and there's no data then read will block, the process will never spawn and we'll still have a `qtile launch` process.

I'm open to suggestions.

Reply:
Perhaps I have PTSD from work, but I think introducing timeouts just masks real bugs. Personally, I'd just do a raw .read() and if that hangs, then we have something to fix.